### ISSUE TITLE
Easy way to exclude PrometheusController from global auth guard?

### ISSUE THREAD

**Issue by mobytoby** [willsoto/nestjs-prometheus#1997]
Hi, first off, thanks for this library!

I have an app that's serving up a couple of endpoints in a Kubernetes environment. I have a global auth guard that's protecting the routes I care about. I'm excepting the public routes from the guard using a `Public()` decorator as detailed in [this blog](https://dev.to/dannypule/exclude-route-from-nest-js-authgaurd-h0). 

I tried passing a custom controller to the prometheus `.register()` function with the `Public()` decorator on it, and I'm not having any luck. 

Before I continue to go down this rabbit hole, I figured I'd ask if there is any obvious (or maybe non-obvious) way to exclude the Prometheus controller from any auth guard in place? I should also add that I'm in no way a nestjs expert. It could be that there's a way to handle this that's more a function of NestJS and less of this particular library. In any event, would love to get your take. Thanks!

**Comment by mobytoby** [comment]
I got around this by using an entirely different path.

In my case, I have a global prefix that's `/*` because this service is deployed to many different routes, each with a different prefix in Kubernetes. For instance `http://some-load-balancer.com/app/api/products` where `app` can really be anything. 

My controller looks something like this:
```
@Controller('api/products')
...
@Get()
fetchAll() { }
...
@Post()
create() {}
```

I wanted each of these services to have their own metrics endpoint, so I deployed your library with a `.register({ path: '/api/products/metrics' })`. It seems that deploying it to the `/products` path was causing the auth-guard to kick in. If I instead just deployed the metrics endpoint to `.register({ path: '/products/metrics' })`, it works fine. The actual endpoint is `http://some-load-balancer/app/products/metrics` and that works perfectly fine. 

Not sure if this is an actual issue worth considering or not (without the workaround). Feel free to close if not. Thanks!
